### PR TITLE
[10-10CG] Add paginated facilities endpoint to CG controller

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): NO
- The front end work that uses this endpoint is behind a flipper toggle, so the backend does not have the toggle. 
- The lighthouse api returns pagination information, but the existing Facilities/V1/Client did not return the full response object. This PR adds a method to the client to return the full response, as well as a new caregivers route that returns the facilities list with pagination metadata. 
- Update client to handle no `data` key being returned by the lighthouse api. This is technically possible if you request a page that is outside of the page range. My forthcoming `vets-website` changes will eliminate this issue from the front end, but the endpoint could technically be called with page 5000 for a result set of 1. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/70846

## Testing done

- [x] *New code is covered by unit tests*
- Added spec for `Facilities/V1/Client`

## What areas of the site does it impact?
10-10CG

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

I updated our existing controller spec for this route. I would eventually like to replace that spec with a request spec, but at this moment I didn't want to take on the scope of that work unless someone really wants me to.

### CODE REVIEW

File: app/controllers/v0/caregivers_assistance_claims_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class CaregiversAssistanceClaimsController < ApplicationController
     before_action :load_user, only: :create
 
     before_action :record_submission_attempt, only: :create
-    before_action :initialize_claim
+    before_action :initialize_claim, only: [:create, :download_pdf]
```
Comment (coope93) [inline app/controllers/v0/caregivers_assistance_claims_controller.rb:14]: The new `facilities` method does not require initializing a claim. These are the only other two exposed routes in this controller. I think this also makes this `initialize_claim` logic more obvious.

File: config/routes.rb [hunk 2]
```diff
@@ -121,9 +121,13 @@
     resource :hca_attachments, only: :create
     resource :form1010_ezr_attachments, only: :create
 
-    resources :caregivers_assistance_claims, only: :create
-    post 'caregivers_assistance_claims/download_pdf', to: 'caregivers_assistance_claims#download_pdf'
-
+    resources :caregivers_assistance_claims, only: :create do
+      collection do
+        get(:facilities)
```
Comment (coope93) [inline config/routes.rb:126]: Added new `facilities` route, and moved the post declaration here. This is functionally the same as before, just easier to read.

File: lib/lighthouse/facilities/v1/response.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def initialize(body, status)
           self.body = body
           self.status = status
           parsed_body = JSON.parse(body)
-          self.data = parsed_body['data']
+          self.data = parsed_body['data'] || []
```
Comment (coope93) [inline lib/lighthouse/facilities/v1/response.rb:24]: If no facilities are returned, the lighthouse api does not return a data key or object. This handles that scenario by setting it to an empty array.

File: spec/controllers/v0/caregivers_assistance_claims_controller_spec.rb [hunk 4]
```diff
@@ -244,4 +245,55 @@
       ).to eq(false)
     end
   end
+
+  describe '#facilities' do
```
Comment (coope93) [inline spec/controllers/v0/caregivers_assistance_claims_controller_spec.rb:249]: As mentioned above, I'd like to eventually move these to request specs.

File: spec/requests/v0/caregivers_assistance_claims_spec.rb [hunk 5]
```diff
@@ -55,4 +56,66 @@
       ).to eq(false)
     end
   end
+
+  describe 'GET /v0/caregivers_assistance_claims/facilities' do
```
Comment (coope93) [inline spec/requests/v0/caregivers_assistance_claims_spec.rb:60]: New request spec

File: spec/controllers/v0/caregivers_assistance_claims_controller_spec.rb (hunk 4 above)
Comment (coope93) [reply spec/controllers/v0/caregivers_assistance_claims_controller_spec.rb]: I deleted these and moved them to the request spec.

**Review by Mitch-A6 — APPROVED** [review]

**Comment by ryan-mcneil** [comment]
@coope93 Please pull master to clear up the CVE

File: lib/lighthouse/facilities/v1/response.rb (hunk 3 above)
Comment (coope93) [reply lib/lighthouse/facilities/v1/response.rb:24]: Yep! It's only an issue with the pagination params with a page that is outside of the total results (ie give me page 10 with 10 per page of a total result set of 5).

**Review by Mitch-A6 — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]

**Review by eselkin — APPROVED** [review]
LGTM

**Pull request merged**